chatbot/src/main/java/chatbot/core/handlers/automaticworkflow/AutomaticWorkflow.java

```diff
@@ -0,0 +1,247 @@
+			responselist.setError();
+			return responselist;
+		}
+
+	}
+	
+	public ResponseList startFunction(String query) {
+
+		String[] splitQuery = query.trim().split(" ");
+		
+		ResponseList returnValue =  new ResponseList();
+		Response responseData = new Response();
+		String extractor = null;
+		String fileName = null;
+
+		if (splitQuery[0].equalsIgnoreCase("extract") ) {
+			if(splitQuery.length==5) {
+				fileName = splitQuery[2];
+				extractor = splitQuery[4];
+			}else {
+				responseData.setContent("The query should be \"Extract From Filename Using Extractor\"");
+				returnValue.addMessage(responseData);
+				return returnValue;
+			}
+			// Future TODO: Make a check for valid Extractor
+			if( !extractor.isEmpty()) {
+				returnValue = checkFilePresent(extractor, fileName);
+									
+			} else {
+				responseData.setContent("Invalid Extractor");
+				returnValue.addMessage(responseData);
+			}
+		}
+
+		return returnValue;		
+	}
+
+	public ResponseList checkFilePresent(String extractor, String fileName) {
+		ResponseList fileInfo =  new ResponseList();
+		Response responseData = new Response();
+				
+		String uri = "http://REPO-MS/getHdfsStructure?location=repo";
+  		String response = restTemplate.getForObject(uri, String.class);
+  		
+  		JsonElement jelement = new JsonParser().parse(response);
+		JsonObject jsonInputText = jelement.getAsJsonObject();
+
+		JsonArray typeValue = jsonInputText.getAsJsonArray("fileList");
+   		List<String> fileList = new ArrayList<String>();
+
+  		for(int i = 0; i<typeValue.size(); i++) {
+  			 JsonObject objectTest = typeValue.get(i).getAsJsonObject();
+  			 String receivedFileName = objectTest.get("suffix").getAsString();
+  			 fileList.add(receivedFileName);
+  		}
+			 if(fileList.contains(fileName)) {
+				 log.info("FILE PRESENT");				
+					fileInfo= constructWorkFlow(extractor,fileName);				
+			 } else {
+				 log.info("FILE NOT PRESENT");
+				 responseData.setContent("Please enter the correct file name");
+				 fileInfo.addMessage(responseData);
+			 }
+
+		return fileInfo; 
+	}
+	
+	public ResponseList constructWorkFlow(String extractor, String fileName) {
+		/*
+		 * create
+		 */
+		// o1		
+		Map<String, String> outputs1 = new HashMap<>();	
+		outputs1.put("output_55uqv2wzcus", "NL");
+		outputs1.put("output_y97vayejxre", "NL");
+
+		Operator o1 = new Operator();
+		o1.setType("file");
+		o1.setId("node_hgibcp02st");
+		o1.setContent("/"+fileName);
+		o1.setOutputs(outputs1);
+		
+		Map<String, Operator> op1 = new HashMap<>();
+		op1.put("node_hgibcp02st", o1);
+		
+		// o2
+		Map<String, String> inputs2 = new HashMap<>();
+		
+		inputs2.put("input_tzil25otre", "NL");
+		inputs2.put("input_2hfqgk71ukp", "NL");
+
+		Map<String, String> outputs2 = new HashMap<>();
+		outputs2.put("output_a2b3epd6nd", "RDF");
+		outputs2.put("output_09phmd5hdmvi", "RDF");
+
+		Operator o2 = new Operator();
+		o2.setType("extractor");
+		o2.setId("node_178rj2s179x");
+		o2.setContent("FOX-MS");

```

Comment:
Hardcoded Extractor, Dataset.  

Will this be a future TODO for next Project group?

Reply:
Yes this is the future TODO and can be added in to enhancement.